data/lib/src/repository/myspace/document_repository_impl.dart

```diff
@@ -31,8 +31,9 @@
 
 import 'package:data/src/datasource/document_datasource.dart';
 import 'package:domain/domain.dart';
+import 'package:domain/src/model/document/document.dart';
 
-class DocumentRepositoryImpl extends DocumentRepository {
+class DocumentRepositoryImpl implements DocumentRepository {
```

Comment:
(curiously) what is the different?

Reply:
its like other `OOP` language. you can only `extend` from only one class and `implements` multi `abstract class`. So we better use `implements` right.